src/dev/lupio/LupioTTY.py

```diff
@@ -37,18 +37,5 @@ class LupioTTY(BasicPioDevice):
     cxx_header = "dev/lupio/lupio_tty.hh"
     terminal = Param.SerialDevice(Parent.any, "The terminal")
     pio_size = Param.Addr(0x1000, "PIO size")
-    platform = Param.Platform(Parent.any,
-                              "Platform this device is part of.")
-
-    def generateDeviceTree(self, state):
-        node = self.generateBasicPioDeviceNode(state, "lupio-tty",
-                                               self.pio_addr,
-                                               self.pio_size)
-        platform = self.platform.unproxy(self)
-        pic = platform.pic
-        node.append(
-            FdtPropertyWords("interrupts", [platform.lupio_tty_int_id]))
-        node.append(
-                FdtPropertyWords("interrupt-parent", state.phandle(pic)))
-        node.appendCompatible(["lupio,tty"])
-        yield node
+    pic = Param.Plic("PLIC")
+    lupio_tty_int_id = Param.Int(0xa, "interrupt")
```

Comment:
```suggestion
    int_id = Param.Int(0xa, "The interrupt number for this device, default works for RISC-V")
```

I'd add some more details to the description. I'm not sure if my details are correct.